profiling_router_test.go

```diff
@@ -0,0 +1,234 @@
+
+		// assert
+		require.NoError(t, err)
+		assert.False(t, isProfilingEnabled(profilingRouter))
+	})
+
+	t.Run("It should not start profiling if not a bool", func(t *testing.T) {
+		// arrange
+		t.Setenv("PPROF_ENABLED", "obviouslyNotABool")
+		ctx := context.Background()
+
+		// act
+		profilingRouter, err := NewProfilingRouter(ctx)
+
+		// assert
+		require.Error(t, err, "fail to initialize pprof profiling: fail to parse environment variable PPROF_ENABLED: strconv.ParseBool: parsing \"obviouslyNotABool\": invalid syntax")
+		assert.False(t, isProfilingEnabled(profilingRouter))
+	})
+
+	t.Run("It should not start profiling if not enabled", func(t *testing.T) {
+		// arrange
+		t.Setenv("PPROF_ENABLED", "false")
+		ctx := context.Background()
+
+		// act
+		profilingRouter, err := NewProfilingRouter(ctx)
+
+		// assert
+		require.NoError(t, err)
+		assert.False(t, isProfilingEnabled(profilingRouter))
+	})
+
+	t.Run("It should not start profiling if username is not set", func(t *testing.T) {
+		// arrange
+		t.Setenv("PPROF_ENABLED", "true")
+		t.Setenv("PPROF_PASSWORD", password)
+		ctx := context.Background()
+
+		// act
+		profilingRouter, err := NewProfilingRouter(ctx)
+
+		// assert
+		require.NoError(t, err)
+		assert.False(t, isProfilingEnabled(profilingRouter))
+	})
+
+	t.Run("It should not start profiling if password is not set", func(t *testing.T) {
+		// arrange
+		t.Setenv("PPROF_ENABLED", "true")
+		t.Setenv("PPROF_USERNAME", username)
+		ctx := context.Background()
+
+		// act
+		profilingRouter, err := NewProfilingRouter(ctx)
+
+		// assert
+		require.NoError(t, err)
+		assert.False(t, isProfilingEnabled(profilingRouter))
+	})
+
+	t.Run("It should start profiling if all environment variables are set", func(t *testing.T) {
+		// arrange
+		t.Setenv("PPROF_ENABLED", "true")
+		t.Setenv("PPROF_USERNAME", username)
+		t.Setenv("PPROF_PASSWORD", password)
+		ctx := context.Background()
+
+		// act
+		profilingRouter, err := NewProfilingRouter(ctx)
+
+		// assert
+		require.NoError(t, err)
+		assert.True(t, isProfilingEnabled(profilingRouter))
+	})
+}
+
+func TestProfilingRouterNotExistingRoute(t *testing.T) {
+	t.Run("it should respond not found", func(t *testing.T) {
+		// arrange
+		path := PprofRoutePrefix + "/path_does_not_exist"
+
+		t.Setenv("PPROF_ENABLED", "true")
+		t.Setenv("PPROF_USERNAME", username)
+		t.Setenv("PPROF_PASSWORD", password)
+
+		ctx := createLog()
+
+		profilingRouter, err := NewProfilingRouter(ctx)
+		require.NoError(t, err)
+
+		httpRecorder := httptest.NewRecorder()
+
+		request := createGetRequest(t, path, username, password)
+
+		// act
+		profilingRouter.ServeHTTP(httpRecorder, request)
+
+		// assert
+		if status := httpRecorder.Code; status != http.StatusNotFound {
+			t.Errorf("handler returned wrong status code: got %v want %v",
+				status, http.StatusNotFound)
+		}
+	})
+}
+
+func TestProfilingRouterEndpoint(t *testing.T) {
+	for _, path := range pathsToTest {
```

Comment:
I would declare the pathsToTest right here rather than at the very top of the file.
They are used only for that test, it would be easier to find those when reading the test